src/Kafka/ManageKafkaPermissions/components/AssignPermissionsManual.tsx

```diff
@@ -0,0 +1,156 @@
+import { Tbody, Tr, Td } from "@patternfly/react-table";
+import { ResourceName } from "./ResourceName";
+import { ResourceOperation, ResourceOperationValue } from "./ResourceOperation";
+import {
+  ResourcePermission,
+  ResourcePermissionValue,
+} from "./ResourcePermission";
+import {
+  ResourcePrefixRule,
+  ResourcePrefixRuleValue,
+} from "./ResourcePrefixRule";
+import { ResourceType, ResourceTypeValue } from "./ResourceType";
+import { ShortcutsTableHead } from "./ShortcutsTableHead";
+import { useTranslation } from "react-i18next";
+import { KafkaInstanceManualPermissions } from "./KafkaInstanceManualPermissions";
+
+export type AssignPermissionsManualProps = {
+  resourceType: ResourceTypeValue | undefined;
+  onChangeResourceType: (value: ResourceTypeValue | undefined) => void;
+  submitted: boolean;
+  resourcePrefix: ResourcePrefixRuleValue;
+  onChangeResourcePrefix: (value: ResourcePrefixRuleValue) => void;
+  resourceName: string | undefined;
+  onChangeResource: (value: string | undefined) => void;
+  onFetchResourceNameOptions: (filter: string) => Promise<string[]>;
+  resourcePermission: ResourcePermissionValue;
+  onChangeResourcePermission: (value: ResourcePermissionValue) => void;
+  resourceOperation: ResourceOperationValue | undefined;
+  onChangeResourceOperation: (
+    value: ResourceOperationValue | undefined
+  ) => void;
+  multipleShorctutPermissions?: boolean;
+};
+
+export const AssignPermissionsManual: React.FC<
+  AssignPermissionsManualProps
+> = ({
+  resourceType,
+  onChangeResourceType,
+  submitted,
+  resourcePrefix,
+  onChangeResourcePrefix,
+  resourceName,
+  onChangeResource,
+  onFetchResourceNameOptions,
+  resourcePermission,
+  resourceOperation,
+  onChangeResourceOperation,
+  onChangeResourcePermission,
+  multipleShorctutPermissions = false,
+}) => {
+  const { t } = useTranslation("manage-kafka-permissions");
+  const resourceTypeOptions = () => {
+    if (resourceType === undefined)
+      return [
```

Comment:
Same code we have in switch default case. why do we have it duplicate ?
Switch default case will manage undefined check.